global/scripts/debug.shlib

```diff
@@ -22,10 +22,10 @@
 #   $3 - exit status
 
 debug_on_error()
-{    
-    if [ ! $HCP_DEBUG_USETRAP == 'OFF' ]; then        
+{
+    if [[ "$HCP_DEBUG_USETRAP" == "ON" ]]; then
```

Comment:
FYI, these edits are from the branch I told him to use.  This PR probably shouldn't go into master, but into the `debug-function-enable` branch.  Once the testing is complete, we can rebase that branch onto master.

Reply:
I do think we want to do this expeditiously, as Misha needs to use the updated structural pipelines for longitudinal pipeline development.

Reply:
Will Misha need all 3 structural pipelines converted before starting?  If so, we should still do them into the `debug-function-enable` branch, test them with that change, and then I can rearrange the commits so the new versions are on master, `debug-function-enable` is based on that, and the rest of the PRs can continue the same way.

Even if he needs each one as soon as they are available, I can do that kind of rearrangement 3 times without much trouble, so that we don't have to test them twice (without, and then with) the debug.shlib changes.

Reply:
I think it would be best if he had all three pipelines converted before doing any development on them.  There is also the longitudinal pipeline.  Perhaps I can convince Jure to convert that one too.